src/rgw/rgw_op.cc

```diff
@@ -6488,14 +6498,27 @@ void RGWAbortMultipart::execute(optional_yield y)
   op_ret = -EINVAL;
   string upload_id;
   upload_id = s->info.args.get("uploadId");
-  rgw_obj meta_obj;
+  std::unique_ptr<rgw::sal::Object> meta_obj;
```

Comment:
why do you use a `uniqe_ptr` here?
is the `meta_obj` allocated here?

Reply:
yes, it's allocated in `get_meta_obj()` method

Reply:
can you declare that inside the "if" statement?
```cpp
std::unique_ptr<meta_obj> = upload->get_meta_obj();
```